terraform/workspaces/infra/postgres/rds.tf

```diff
@@ -84,12 +90,12 @@ resource "aws_db_instance" "postgres" {
 
   allocated_storage           = 20
   max_allocated_storage       = 1000
-  allow_major_version_upgrade = true
+  allow_major_version_upgrade = false
   auto_minor_version_upgrade  = true
   availability_zone           = var.availability_zones.names[0]
   backup_retention_period     = 7
   monitoring_interval         = 15
-  multi_az                    = false
+  multi_az                    = var.multi_az_enabled
```

Comment:
`availability_zone` needs a guard when `multi_az_enabled` is true otherwise it yields the following error:

```
│ Error: creating RDS DB Instance (paragon-enterprise-yegcszqd): InvalidParameterCombination: Requesting a specific availability zone is not valid for Multi-AZ instances.
│ 	status code: 400, request id: 226f4422-8cdd-46a8-97c8-be39f22510c6
│
│   with module.postgres.aws_db_instance.postgres["paragon"],
│   on postgres/rds.tf line 75, in resource "aws_db_instance" "postgres":
│   75: resource "aws_db_instance" "postgres" {

```